homeassistant/components/template/event.py

```diff
@@ -0,0 +1,270 @@
+) -> list[TemplateEvent]:
+    """Create the Template event."""
+    entities = []
+    for definition in definitions:
+        unique_id = definition.get(CONF_UNIQUE_ID)
+        if unique_id and unique_id_prefix:
+            unique_id = f"{unique_id_prefix}-{unique_id}"
+        entities.append(TemplateEvent(hass, definition, unique_id))
+    return entities
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the template select."""
+    if discovery_info is None:
+        _LOGGER.warning(
+            "Template event entities can only be configured under template:"
+        )
+        return
+
+    if "coordinator" in discovery_info:
+        async_add_entities(
+            TriggerEventEntity(hass, discovery_info["coordinator"], config)
+            for config in discovery_info["entities"]
+        )
+        return
+
+    async_add_entities(
+        await _async_create_entities(
+            hass, discovery_info["entities"], discovery_info["unique_id"]
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Initialize config entry."""
+    _options = dict(config_entry.options)
+    _options.pop("template_type")
+    validated_config = EVENT_CONFIG_SCHEMA(_options)
+    async_add_entities([TemplateEvent(hass, validated_config, config_entry.entry_id)])
+
+
+class AbstractTemplateEvent(EventEntity):
+    """Representation of a template event features."""
+
+    def __init__(self, config: dict[str, Any]) -> None:
+        """Initialize the features."""
+        self._event_type_template = config[CONF_EVENT_TYPE]
+        self._event_types_template = config[CONF_EVENT_TYPES]
+
+        self._attr_device_class = config.get(CONF_DEVICE_CLASS)
+
+        self._event_type = None
+        self._attr_event_types = []
+
+    @property
+    def event_types(self) -> list[str]:
+        """Return the event_types list."""
+        return self._attr_event_types
+
+    @callback
+    def _update_event_types(self, event_types: Any) -> None:
+        """Update the event types from the template."""
+        if event_types in (None, "None", ""):
+            self._attr_event_types = []
+            return
+
+        if not isinstance(event_types, list):
+            _LOGGER.error(
+                (
+                    "Received invalid event_types list: %s for entity %s. Expected list of"
+                    " strings"
+                ),
+                event_types,
+                self.entity_id,
+            )
+            self._attr_event_types = []
+            return
+
+        if len(event_types) == 0:
+            self._attr_event_types = []
+            return
+
+        self._attr_event_types = event_types
```

Comment:
Isn't this practically doing the same as they get assigned an empty list in both cases

Reply:
it is, I'll update